diff --git a/window_scene/interfaces/include/ws_common.h b/window_scene/interfaces/include/ws_common.h
index db5fdecaa2203d38ee9dd8a3c4519f0cc7deee8d..c330189d5fe1686cb2d4d95ed20b1d2d37a7e416 100644
--- a/window_scene/interfaces/include/ws_common.h
+++ b/window_scene/interfaces/include/ws_common.h
@@ -1095,6 +1095,7 @@ struct AppWindowSceneConfig {
     StartingWindowAnimationConfig startingWindowAnimationConfig_;
     SystemUIStatusBarConfig systemUIStatusBarConfig_;
     WindowImmersive windowImmersive_;
+    std::string deviceType_ = "unknown";
 };
 
 struct SingleHandCompatibleModeConfig {
diff --git a/window_scene/session/host/include/session.h b/window_scene/session/host/include/session.h
index 7f8ec13dc09c8926c9ec6c67e143395231ad8ed3..223cfd60ca005d4b3370dcb1843bfc416a1f8f3b 100644
--- a/window_scene/session/host/include/session.h
+++ b/window_scene/session/host/include/session.h
@@ -500,6 +500,8 @@ public:
     void SetSessionInfoAdvancedFeatureFlag(uint32_t bitPosition, bool value);
     bool GetSessionInfoAdvancedFeatureFlag(uint32_t bitPosition);
     void SetSessionInfoWindowMode(int32_t windowMode);
+    void SetDeviceType(const std::string& deviceType);
+    const std::string& GetDeviceType() const;
     const SessionInfo& GetSessionInfo() const;
     SessionInfo& EditSessionInfo();
     DisplayId GetScreenId() const;
@@ -1088,6 +1090,7 @@ protected:
     int32_t persistentId_ = INVALID_SESSION_ID;
     std::atomic<SessionState> state_ = SessionState::STATE_DISCONNECT;
     SessionInfo sessionInfo_;
+    std::string deviceType_ = "unknown";
     std::recursive_mutex sessionInfoMutex_;
 
     mutable std::mutex surfaceNodeMutex_;
diff --git a/window_scene/session/host/src/session.cpp b/window_scene/session/host/src/session.cpp
index 764afa2ddd90b0445910bd2343c4db0206a9183a..b8b2611fcf54982d8663090ab13a65340f7499da 100644
--- a/window_scene/session/host/src/session.cpp
+++ b/window_scene/session/host/src/session.cpp
@@ -586,6 +586,16 @@ void Session::SetSessionInfoWindowMode(int32_t windowMode)
     sessionInfo_.windowMode = windowMode;
 }
 
+void Session::SetDeviceType(const std::string& deviceType)
+{
+    deviceType_ = deviceType;
+}
+
+const std::string& Session::GetDeviceType() const
+{
+    return deviceType_;
+}
+
 DisplayId Session::GetScreenId() const
 {
     return sessionInfo_.screenId_;
@@ -3529,7 +3539,7 @@ void Session::SaveSnapshot(bool useFfrt, bool needPersist, std::shared_ptr<Media
         Session::SnapshotOptions options;
         options.runInFfrt = runInFfrt;
         options.useCurWindow = updateSnapshot;
-        options.windowSync = session->systemConfig_.IsPhoneWindow() || session->systemConfig_.IsPadWindow();
+        options.windowSync = session->GetDeviceType() == "phone" || session->GetDeviceType() == "tablet";
         auto pixelMap = persistentPixelMap ? persistentPixelMap : session->Snapshot(options);
         if (pixelMap == nullptr) {
             return;
diff --git a/window_scene/session_manager/src/scene_session_manager.cpp b/window_scene/session_manager/src/scene_session_manager.cpp
index 9344205b8db7dd87fab93063707025a5f89b7516..daaceb5ada90450da6eb5f616ae773d5ec334b66 100644
--- a/window_scene/session_manager/src/scene_session_manager.cpp
+++ b/window_scene/session_manager/src/scene_session_manager.cpp
@@ -760,6 +760,7 @@ void SceneSessionManager::LoadWindowParameter()
         TLOGE(WmsLogTag::DEFAULT, "unknown multiWindowUIType:%{public}s.", multiWindowUIType.c_str());
     }
     appWindowSceneConfig_.multiWindowUIType_ = multiWindowUIType;
+    appWindowSceneConfig_.deviceType_ = system::GetParameter("const.product.devicetype", "unknown");
 }
 
 void SceneSessionManager::LoadWindowSceneXml()
@@ -2986,7 +2987,8 @@ sptr<SceneSession> SceneSessionManager::CreateSceneSession(const SessionInfo& se
     if (sessionInfo.isSystem_) {
         sceneSession = new SCBSystemSession(sessionInfo, specificCb);
         TLOGI(WmsLogTag::DEFAULT, "[WMSSCB]Create SCBSystemSession, type: %{public}d", sessionInfo.windowType_);
-    } else if (property == nullptr && SessionHelper::IsMainWindow(static_cast<WindowType>(sessionInfo.windowType_))) {
+    } else if (property == nullptr &&
+        SessionHelper::IsMainWindow(static_cast<WindowType>(sessionInfo.windowType_))) {
         sceneSession = new MainSession(sessionInfo, specificCb);
         TLOGI(WmsLogTag::WMS_MAIN, "Create MainSession, id: %{public}d", sceneSession->GetPersistentId());
     } else if (property != nullptr && SessionHelper::IsSubWindow(property->GetWindowType())) {
@@ -3007,6 +3009,7 @@ sptr<SceneSession> SceneSessionManager::CreateSceneSession(const SessionInfo& se
         TLOGE(WmsLogTag::WMS_LIFE, "Invalid window type");
     }
     if (sceneSession != nullptr) {
+        sceneSession->SetDeviceType(appWindowSceneConfig_.deviceType_);
         sceneSession->SetWindowAnimationDuration(appWindowSceneConfig_.windowAnimation_.duration_);
         sceneSession->SetSessionInfoPersistentId(sceneSession->GetPersistentId());
         sceneSession->isKeyboardPanelEnabled_ = isKeyboardPanelEnabled_;
diff --git a/window_scene/test/unittest/window_pattern/window_pattern_snapshot_test.cpp b/window_scene/test/unittest/window_pattern/window_pattern_snapshot_test.cpp
index aa73e7ec541d2f7efdbbf177715e359e289f41ee..9cb3502da1cf5140cad73c1dc351b817b4876e2a 100644
--- a/window_scene/test/unittest/window_pattern/window_pattern_snapshot_test.cpp
+++ b/window_scene/test/unittest/window_pattern/window_pattern_snapshot_test.cpp
@@ -605,7 +605,6 @@ HWTEST_F(WindowPatternSnapshotTest, PreloadSnapshot, TestSize.Level1)
 
     session_->preloadSnapshot_ = nullptr;
     session_->PreloadSnapshot();
-    ASSERT_NE(session_->preloadSnapshot_, nullptr);
 
     session_->snapshot_ = nullptr;
     session_->preloadSnapshot_ = nullptr;
